dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/HostStatusTracker.java

```diff
@@ -128,9 +128,45 @@ public boolean inactiveSetChanged(Collection<Host> hostsUp, Collection<Host> hos
 	 * @return true/false indicating whether the set of hosts has changed or not.
 	 */
 	public boolean checkIfChanged(Collection<Host> hostsUp, Collection<Host> hostsDown) {
-		return activeSetChanged(hostsUp) || inactiveSetChanged(hostsUp, hostsDown);
+		boolean changed = activeSetChanged(hostsUp) || inactiveSetChanged(hostsUp, hostsDown);
+		if (changed) logChangedHosts(hostsUp, hostsDown);
+		return changed;
 	}
-	
+
+	/**
+	 * Helper method that checks if anything has changed b/w the current state and the new set of hosts up and down and
+	 * logs the difference in hosts.
+	 * @param hostsUp
+	 * @param hostsDown
+	 */
+	public void logChangedHosts(Collection<Host> hostsUp, Collection<Host> hostsDown) {
+		if (!logger.isDebugEnabled()) return;
+
+		Set<Host> changedHostsUp = new HashSet<>();
+		changedHostsUp.addAll(hostsUp);
+		for (Host host : activeHosts) {
```

Comment:
Will Java Set operations like https://docs.oracle.com/javase/7/docs/api/java/util/Set.html#removeAll makes this more terse? Few other operations that may help you here are retainAll/containsAll.

Reply:
Thanks! I've incorporated this to make it shorter.